index.js

```diff
@@ -94,7 +94,26 @@ class ServerlessFullstackPlugin {
             });
     }
 
+    setEnvironmentVariables() {
+        this.serverless.cli.log('Setting the environment variables');
+        const environment = this.serverless.service.provider.environment;
+
+        if (!environment) {
+          return this.serverless.cli.log(
+            'No environment variables detected. Skipping step...'
+          );
+        }
+
+        Object.keys(environment).forEach(variable => {
+          this.serverless.cli.log(
+            `Setting ${variable} to ${environment[variable]}`
+          );
+          process.env[variable] = environment[variable];
+        });
```

Comment:
It would be better to build a new object with the vars for the client generation than to potentially clober the environment for the currently running process (serverless).

something like

```js
const serverlessEnv = this.serverless.service.provider.environment;
const clientGenEnv = Object.assign({}, process.env, serverlessEnv);
...
 const proc = spawn(command, args, {cwd: clientSrcPath, env: clientGenEnv, shell: true});
```

Reply:
Hi @andrewphahn, updated the PR as advised. :)